### PR TITLE
Add method change_attractor

### PR DESCRIPTION
**Pull request by jorgepiloto**
Tries to solve #676. Just changed syntax from gist so it can act as an Orbit method. I also included a test based on iss from the examples module.

### CODE REVIEW

**Comment by Juanlu001** [comment]
Thanks for the PR @jorgepiloto! I see one danger with this:

The original intent of the function was to change the attractor for the Galilean moons from the Sun to Jupiter. This makes sense, because the real attractor of these moons is Jupiter. In fact, if one takes the data of `.from_horizons` and just `.propagate`, the results are completely wrong, because the propagation happens as if Jupiter does not even exist.

I think having the possibility to change the attractor of any orbit in this way is quite dangerous. I should have thought about the problem when writing the issue.

What do you think about making this method a private function in `poliastro.twobody.orbit` and looking for a way to fix the mismatch of `from_horizons`?

**Comment by jorgepiloto** [comment]
Sure! Therefore, should we add something like:

```python
```

**Comment by jorgepiloto** [comment]
Sure! What do you think about something like this?

```python
ss = Orbit.from_horizons(name, attractor=Sun, **kwargs)
```

**Comment by Juanlu001** [comment]
Looking at https://astroquery.readthedocs.io/en/latest/api/astroquery.jplhorizons.HorizonsClass.html, I don't think `from_horizons` can _know_ which attractor should it use automatically. So yes, there should be a way for the user to specify that.

After thinking about your proposal a bit more (see thoughts in #676), I think `Orbit.from_horizons(name).change_attractor(Jupiter)` is a good idea. However, perhaps we could check wether the object falls inside the Sphere of Influence of the new attractor (not sure whether Laplace or Hill) and, if not, display a warning or even raise an error. What do you think @jorgepiloto?

**Comment by jorgepiloto** [comment]
It would be very easy to check by using the `laplace_radius` in `poliastro.threebody.soi` if the assumption of patched conics is suitable.

The problem comes when the orbit is under this sphere of influence but its nature is parabolic or hyperbolic (flyby case). The algorithm will change the attractor, but if the user tries to propagate, there would be some point at which computations are no longer correct and attractor should be changed again, as you said in  [your comment](https://github.com/poliastro/poliastro/issues/676#issuecomment-497959261).

Knowing the point at which the previous situation occurs is easy but implement that inside the propagators methods I think may be crazy...

We could enable this method to work if two conditions are fulfilled:
* We are under the sphere of influence
* Orbit of the body is elliptical or circular (closed orbit) 

Otherwise, we raise an error for the moment. Could this be a good approach?

**Comment by Juanlu001** [comment]
> Knowing the point at which the previous situation occurs is easy but implement that inside the propagators methods I think may be crazy...

Actually, now that [SciPy propagators have support for events](https://docs.scipy.org/doc/scipy/reference/generated/scipy.integrate.solve_ivp.html), it's not that difficult :wink: but let's leave that for a future issue.

I agree with raising an error if the orbit is outside the sphere of influence of the target attractor with respect to the Sun. However, let's not consider the eccentricity here, to allow open orbits as well.

**Comment by Juanlu001** [comment]
Wow, investigating these failures.

**Comment by Juanlu001** [comment]
Oops, we introduced a circular import:

```
  File "/home/circleci/repo/venv/lib/python3.7/site-packages/poliastro/twobody/__init__.py", line 1, in <module>
    from .orbit import Orbit
  File "/home/circleci/repo/venv/lib/python3.7/site-packages/poliastro/twobody/orbit.py", line 22, in <module>
    from poliastro.threebody.soi import laplace_radius
  File "/home/circleci/repo/venv/lib/python3.7/site-packages/poliastro/threebody/soi.py", line 32, in <module>
    from poliastro.twobody import Orbit
```

**Comment by Juanlu001** [comment]
The `Orbit` object is used in the SOI module to... Compute the classical elements of the planets!

https://github.com/jorgepiloto/poliastro/blob/dbbf332386e135eb64d0bf62d114350deb5c072d/src/poliastro/threebody/soi.py#L54-L55

This leads me to two things:

* My dreaded `from_body_ephem` strikes back https://github.com/poliastro/poliastro/issues/445 (see some comments at the end about sbpy)
* We had a long discussion with @AunSiro about adding some "mean" elements to the `Body` classes that we could use for this kind of thing https://github.com/poliastro/poliastro/pull/153

So my solution to this circular import is to make the `Orbit` import in `poliastro.threebody.soi` "delayed" while we figure out a better way.

File: src/poliastro/tests/tests_twobody/test_orbit.py [hunk 1]
```diff
@@ -915,3 +916,9 @@ def test_orbit_wrong_dimensions_fails():
     with pytest.raises(ValueError) as excinfo:
         Orbit.from_vectors(Earth, bad_r, bad_v)
     assert "ValueError: Vectors must have dimension 1, got 2 and 3" in excinfo.exconly()
+
+
+def test_orbit_change_attractor_error():
+    with pytest.raises(ValueError) as excinfo:
+        iss.change_attractor(Sun)
```
Comment (Juanlu001) [inline src/poliastro/tests/tests_twobody/test_orbit.py]: It turns out that computing the radius of the sphere of influence of the Sun makes no sense :)

> In the patched conic approximation, once an object leaves the planet's SOI, the primary/only gravitational influence is the Sun (until the object enters another body's SOI).

https://www.wikiwand.com/en/Sphere_of_influence_(astrodynamics)

I think we need two kinds of errors:

* If the attractor is set to the `Sun` and we do `set_attractor(new_attractor)`, we check the SOI of `new_attractor` and we raise the error you wrote here (`"ValueError: Body is out of new attractor's SOI."`)
* If the attractor is set to `attractor` and we do `set_attractor(attractor.parent)`, we check the SOI of `attractor` and, _if we are still inside the SOI of `attractor`_:
  - If the orbit is closed, raise `"ValueError: Body will never leave the SOI of its current attractor"`
  - If the orbit is open, just do it! (I think emitting a warning here might be too noisy, but we could as well say `"PatchedConicsWarning: Leaving the SOI of the current attractor"`

What do you think @jorgepiloto?

Comment (jorgepiloto) [reply src/poliastro/tests/tests_twobody/test_orbit.py]: I think raising these 2 errors and the warning, depending on the case, it is very useful. Hope to have this in a couple of minutes :+1:

**Comment by Juanlu001** [comment]
I made several changes to your code:

* Made it more general: instead of checking for the `Sun`, I'm checking for `attractor.parent` (in case people want to define their own planetary systems)
* Removed the capability to change the plane to simplify the code (probably we should have another method for that)
* Implement a short-circuit in case the target and current attractor are the same
* Allow only changes from attractor to parent and vice-versa

**Comment by Juanlu001** [comment]
Aaaaaand merging! Thanks @jorgepiloto to get the ball rolling!

**Pull request merged**